Apigee/SmartDocs/Revision.php

```diff
@@ -401,12 +409,28 @@ public function delete($revisionId = null)
      */
     public function importSwagger($swaggerUrl)
     {
+        $combinedString = rawurlencode($this->apiId) . '/revisions?action=import&format=swagger';
         $this->blankValues();
-        $this->post('?action=import&format=swagger', 'URL=' . $swaggerUrl, 'text/plain; charset=utf-8');
+        $this->post($combinedString, 'URL=' . $swaggerUrl, 'text/plain; charset=utf-8');
         $response = $this->responseObj;
         self::fromArray($this, $response);
     }
 
+  /**
+   * Imports a model revision from a WADL document.
+   *
+   * @param string $modelId
+   * @param string $xml
+   */
+  public function importSwaggerFile($yaml)
```

Comment:
Does this work?  With the other changes I made above, this didn't seem like the obvious way to do things.  I would be happy to discuss further.